### PR TITLE
Add demo: Conway's game of life

### PR DESCRIPTION
**Pull request by HypnosNova**
image

### CODE REVIEW

**Comment by austinEng** [comment]
Ah sorry, I completely missed this. Could you rebase it and I'll take a look?

File: src/sample/gameOfLife/main.ts [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+    requestAnimationFrame(loop);
+  })();
+};
+
+const GameOfLife: () => JSX.Element = () =>
+  makeSample({
+    name: "Conway's Game of Life",
+    description:
+      "This example shows how to make Conway's game of life by using compute pipeline and render pipeline.",
```
Comment (austinEng) [inline src/sample/gameOfLife/main.ts:279]: overall, would be great to add a small description here about how it works - like cell positions ping pong between two storage buffers. and you use those cell positions to draw square sprites.

and then a few comments throughout the code about what you're doing (making sprite vertex buffer, etc.)

**Review by austinEng — COMMENTED** [review]
a few small comments, but otherwise looks great!

File: src/sample/gameOfLife/main.ts [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+      {
+        shaderLocation: 0,
+        offset: 0,
+        format: 'uint32',
+      },
+    ],
+  };
+
+  function addGUI() {
+    gui.add(GameOptions, 'speed', 0, 60, 1);
```
[the remark this answers is not in the document]
Comment (HypnosNova) [reply src/sample/gameOfLife/main.ts]: handled

**Comment by HypnosNova** [comment]
> looks

Thanks a lot. All codes are improved

**Comment by HypnosNova** [comment]
> a few small comments, but otherwise looks great!

Thanks a lot. All codes have been improved

**Review by austinEng — APPROVED** [review]

**Pull request merged**